lib/liquid/tags/render.rb

```diff
@@ -107,18 +127,29 @@ def rigid_parse(markup)
       p.consume?(:comma)
 
       @attributes = {}
-      while p.look(:id)
-        key = p.consume
-        p.consume(:colon)
-        @attributes[key] = safe_parse_expression(p)
+      while p.look(:dotdotdot) || p.look(:id)
+        if p.consume?(:dotdotdot)
+          @attributes.delete("...")
+          @attributes["..."] = true
```

Comment:
I think we have to be thinking that we might expand this to `...product`. If we do, it would probably mean that we'd implement this differently. 

We need some kind of system for "spreading" props of something. Even if right now we're only spreading the props of the global context.

Reply:
Personally, I think something like this would be good enough for now (then we may expand the error space in the future)

```ruby
...
@attributes = {}
while p.look(:dotdotdot) || p.look(:id)
  if p.consume?(:dotdotdot)
    raise SyntaxError if !p.look(:comma) && !p.look(:end_of_string) # TODO: error message
    @attributes.delete("...")
    @attributes["..."] = true
  else
    key = p.consume
    p.consume(:colon)
    @attributes.delete(key)
    @attributes[key] = safe_parse_expression(p)
  end
  p.consume?(:comma) # optional comma
end

p.consume(:end_of_string)
...
```

What do you think @dejmedus @charlespwd ?

Reply:
Alright so I think all of my comments were unwarranted. Got confused because I saw a hash, but since we're doing `@attributes.each` and since hashes maintain insertion order, we're good with the code as is.

Am a bit worried about the performance of copying all the attributes to make the scoping work, but it's ok.